pkg/service/roommanager.go

```diff
@@ -361,116 +361,127 @@ func (r *RoomManager) rtcSessionWorker(room *rtc.Room, participant types.Partici
-				if err := room.UpdateSubscriptions(participant, msg.Subscription.TrackSids, msg.Subscription.Subscribe); err != nil {
-					logger.Warnw("could not update subscription", err,
-						"room", room.Room.Name,
-						"participant", participant.Identity(),
-						"pID", participant.ID(),
-						"tracks", msg.Subscription.TrackSids,
-						"subscribe", msg.Subscription.Subscribe)
-				}
-			case *livekit.SignalRequest_TrackSetting:
-				for _, sid := range msg.TrackSetting.TrackSids {
-					subTrack := participant.GetSubscribedTrack(sid)
-					if subTrack == nil {
-						logger.Warnw("unable to find SubscribedTrack", nil,
-							"room", room.Room.Name,
-							"participant", participant.Identity(),
-							"pID", participant.ID(),
-							"track", sid)
-						continue
-					}
-
-					// find the source PublishedTrack
-					publisher := room.GetParticipant(subTrack.PublisherIdentity())
-					if publisher == nil {
-						logger.Warnw("unable to find publisher of SubscribedTrack", nil,
-							"room", room.Room.Name,
-							"participant", participant.Identity(),
-							"pID", participant.ID(),
-							"publisher", subTrack.PublisherIdentity(),
-							"track", sid)
-						continue
-					}
-
-					pubTrack := publisher.GetPublishedTrack(sid)
-					if pubTrack == nil {
-						logger.Warnw("unable to find PublishedTrack", nil,
-							"room", room.Room.Name,
-							"participant", publisher.Identity(),
-							"pID", publisher.ID(),
-							"track", sid)
-						continue
-					}
-					if msg.TrackSetting.Width > 0 {
-						msg.TrackSetting.Quality = pubTrack.GetQualityForDimension(msg.TrackSetting.Width, msg.TrackSetting.Height)
-					}
-
-					// find quality for published track
-					logger.Debugw("updating track settings",
-						"room", room.Room.Name,
-						"participant", participant.Identity(),
-						"pID", participant.ID(),
-						"settings", msg.TrackSetting)
-					subTrack.UpdateSubscriberSettings(
-						!msg.TrackSetting.Disabled,
-						msg.TrackSetting.Quality,
-					)
-				}
-			case *livekit.SignalRequest_Leave:
-				_ = participant.Close()
+					"publisher", subTrack.PublisherIdentity(),
+					"track", sid)
+				continue
+			}
+
+			pubTrack := publisher.GetPublishedTrack(sid)
+			if pubTrack == nil {
+				logger.Warnw("unable to find PublishedTrack", nil,
+					"room", room.Room.Name,
+					"participant", publisher.Identity(),
+					"pID", publisher.ID(),
+					"track", sid)
+				continue
 			}
+
+			// find quality for published track
+			logger.Debugw("updating track settings",
+				"room", room.Room.Name,
+				"participant", participant.Identity(),
+				"pID", participant.ID(),
+				"settings", msg.TrackSetting)
+			subTrack.UpdateSubscriberSettings(msg.TrackSetting)
+		}
+	case *livekit.SignalRequest_UpdateLayers:
+		track := participant.GetPublishedTrack(msg.UpdateLayers.TrackSid)
+		if track == nil {
+			logger.Warnw("could not find published track", nil,
+				"track", msg.UpdateLayers.TrackSid)
+
```

Comment:
Maybe the diff is messed up here or I am misunderstanding. Is this supposed to call something on `track` if it is not `nil`?

Reply:
hah, it was late and I had a brain fart. good call!